includes/fraud-prevention/class-buyer-fingerprinting-service.php

```diff
@@ -62,12 +62,21 @@ public function hash_data_for_fraud_prevention( string $data ): string {
 	 * @return array An array of hashed data for an order.
 	 */
 	public function get_hashed_data_for_customer( $fingerprint ): array {
+		$order_items_count = WC()->cart ? intval( WC()->cart->get_cart_contents_count() ) : null;
+		$order_id          = isset( $_POST['wcpay_order_id'] ) ? intval( $_POST['wcpay_order_id'], 10 ) : 0; // phpcs:ignore WordPress.Security.NonceVerification
+		if ( ! $order_items_count && 0 < $order_id ) {
+			$order = wc_get_order( $order_id );
+			if ( $order ) {
+				$order_items_count = $order->get_item_count();
+			}
+		}
+
 		return array_filter(
 			[
 				'fraud_prevention_data_shopper_ip_hash' => $this->hash_data_for_fraud_prevention( WC_Geolocation::get_ip_address() ),
 				'fraud_prevention_data_shopper_ua_hash' => $fingerprint,
 				'fraud_prevention_data_ip_country'      => WC_Geolocation::geolocate_ip( '', true )['country'],
-				'fraud_prevention_data_cart_contents'   => WC()->cart ? intval( WC()->cart->get_cart_contents_count() ) : null,
+				'fraud_prevention_data_cart_contents'   => $order_items_count,
```

Comment:
What happens if we get through all the checks above and `$order_items_count` still ends up being `null` here?

Reply:
This is the case I'm trying to solve here, as the Pay for Order pages don't have a cart object available. We're trying to get it from the order, but it looks like it doesn't exist either.

Reply:
Based on your changes and my testing yesterday, you've got this sorted out for all forms of UPE as well as non-UPE. Marking as resolved.